### PR TITLE
Publish pointclouds with unique pointers

### PR DESCRIPTION
**Pull request by georgflick**
Test together with PR in main repo

Changes:
- Set SensorDataQoS depth to 1
- Publish PointCloud using unique pointers

### CODE REVIEW

File: include/patchworkpp/patchworkpp.hpp [hunk 1]
```diff
@@ -970,21 +970,22 @@ void PatchWorkpp<PointT>::callbackCloud(const sensor_msgs::msg::PointCloud2::Con
...
-    cloud_ROS.header.stamp = stamp;
-    cloud_ROS.header.frame_id = frame_id;
-    return cloud_ROS;
+template<typename PointT> inline
+sensor_msgs::msg::PointCloud2::UniquePtr PatchWorkpp<PointT>::cloud2msg(pcl::PointCloud<PointT> cloud, const rclcpp::Time& stamp, std::string frame_id)
+{
+    sensor_msgs::msg::PointCloud2::UniquePtr cloud_ROS = std::make_unique<sensor_msgs::msg::PointCloud2>();
+    pcl::toROSMsg(cloud, *cloud_ROS);
+    cloud_ROS->header.stamp = stamp;
+    cloud_ROS->header.frame_id = frame_id;
```
Comment (adivardi) [inline include/patchworkpp/patchworkpp.hpp:986]: I think `toROSMsg` set the stamp and header correctly already

**Review by adivardi — CHANGES_REQUESTED** [review]

File: include/patchworkpp/patchworkpp.hpp (hunk 1 above)
Comment (georgflick) [reply include/patchworkpp/patchworkpp.hpp:986]: The header stamp and frame_id are empty in `pcl::PointCloud<PointT> cloud` and that's why it needs to be set here.

File: include/patchworkpp/patchworkpp.hpp [hunk 2]
```diff
@@ -968,23 +968,24 @@ void PatchWorkpp<PointT>::callbackCloud(const sensor_msgs::msg::PointCloud2::Con
     }
 
     const auto node_graph = this->get_node_graph_interface();
```
Comment (adivardi) [inline include/patchworkpp/patchworkpp.hpp]: node_graph can be removed

File: include/patchworkpp/patchworkpp.hpp (hunk 1 above)
Comment (adivardi) [reply include/patchworkpp/patchworkpp.hpp:986]: oh, ok, that's fine then

**Review by adivardi — CHANGES_REQUESTED** [review]

File: include/patchworkpp/patchworkpp.hpp (hunk 2 above)
Comment (georgflick) [reply include/patchworkpp/patchworkpp.hpp]: done

**Review by adivardi — COMMENTED** [review]
code looks fine, Need to test with main repo PR

**Review by adivardi — APPROVED** [review]

**Pull request merged**